Riot/Modules/People/PeopleViewController.m

```diff
@@ -92,302 +76,36 @@ - (void)didReceiveMemoryWarning
-
-- (Class<MXKCellRendering>)cellViewClassForCellData:(MXKCellData*)cellData
-{
-    if ([cellData isKindOfClass:MXKContact.class])
-    {
-        return ContactTableViewCell.class;
-    }
-    
-    return [super cellViewClassForCellData:cellData];
-}
-
-#pragma mark - UITableView data source
-
-- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
-{
-    // Retrieve the current number of sections related to the direct rooms.
-    // Sanity check: check whether the recents data source is correctly configured.
-    directRoomsSectionNumber = 0;
-    
-    if (recentsDataSource.recentsDataSourceMode == RecentsDataSourceModePeople)
-    {
-        directRoomsSectionNumber = [self.dataSource numberOfSectionsInTableView:self.recentsTableView];
-    }
-    
-    // Retrieve the current number of sections related to the contacts
-    contactsSectionNumber = [contactsDataSource numberOfSectionsInTableView:self.recentsTableView];
-    
-    return (directRoomsSectionNumber + contactsSectionNumber);
-}
-
-- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
-{
-    NSInteger count = 0;
-    
-    if (section < directRoomsSectionNumber)
-    {
-        count = [self.dataSource tableView:tableView numberOfRowsInSection:section];
-    }
-    else
-    {
-        section -= directRoomsSectionNumber;
-        if (section < contactsSectionNumber)
-        {
-            count = [contactsDataSource tableView:tableView numberOfRowsInSection:section];
-        }
-    }
-    
-    return count;
-}
-
-- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
-{
-    NSInteger section = indexPath.section;
-    
-    if (section < directRoomsSectionNumber)
-    {
-        return [self.dataSource tableView:tableView cellForRowAtIndexPath:indexPath];
-    }
-    else
-    {
-        section -= directRoomsSectionNumber;
-        if (section < contactsSectionNumber)
-        {
-            return [contactsDataSource tableView:tableView cellForRowAtIndexPath:[NSIndexPath indexPathForRow:indexPath.row inSection:section]];
-        }
-    }
-    
-    // Return a fake cell to prevent app from crashing.
-    return [[UITableViewCell alloc] init];
-}
-
-- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath
-{
-    NSInteger section = indexPath.section;
-    
-    if (section < directRoomsSectionNumber)
-    {
-        return [self.dataSource tableView:tableView canEditRowAtIndexPath:indexPath];
-    }
-    else
-    {
-        section -= directRoomsSectionNumber;
-        if (section < contactsSectionNumber)
-        {
-            return [contactsDataSource tableView:tableView canEditRowAtIndexPath:[NSIndexPath indexPathForRow:indexPath.row inSection:section]];
-        }
-    }
-    
-    return NO;
-}
-
 #pragma mark - UITableView delegate
 
 - (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section
```

Comment:
Should we remove this method to rely on parent class implementation or should we keep returning 0 like `RoomsViewController`?